### PR TITLE
MM-13604 timezone for post dates

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Sets the user-configured timezone -- if any -- on the permalink hover and date separators in the post view.

This also fixes a deprecation warning from jest when running in watch mode.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-13604

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: components/local_date_time/local_date_time.jsx [hunk 1]
```diff
@@ -39,13 +40,18 @@ export default class LocalDateTime extends React.PureComponent {
...
+        }
+
         const timezoneProps = enableTimezone && timeZone ? {timeZone} : {};
 
         return (
             <time
                 className='post__time'
                 dateTime={date.toISOString()}
-                title={date}
+                title={enableTimezone ? title.toString() + ' (' + title.tz() + ')' : title.toString()}
```
Comment (lieut-data) [inline components/local_date_time/local_date_time.jsx]: This formatting is meant to mimic the title formatting today, adding the "friendly" timezone in parentheses after the offset.

Comment (saturninoabril) [reply components/local_date_time/local_date_time.jsx]: (a) Not sure what is expected but currently the timezone in parenthesis on me is "Philippine Standard Time" compared to this change of "Asia/Manila".

(b) If props.enableTimezone is true and props.userTimezone is not set (or undefined), text on parenthesis is showing "(undefined)".  Maybe need another condition check,
```suggestion
                title={enableTimezone && title.tz() ? title.toString() + ' (' + title.tz() + ')' : title.toString()}
```
screen shot 2019-03-06 at 7 15 45 pm

File: components/local_date_time/local_date_time.test.jsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                useMilitaryTime={true}
+                enableTimezone={true}
+                timeZone={'Australia/Sydney'}
+            />
+        );
+        expect(wrapper).toMatchSnapshot();
+
+        expect(wrapper.find('span').text()).toBe('15:15');
+    });
+});
```
Comment (saturninoabril) [inline components/local_date_time/local_date_time.test.jsx]: 💯

File: package.json [hunk 3]
```diff
@@ -161,7 +161,9 @@
     "setupFiles": [
       "jest-canvas-mock"
     ],
-    "setupTestFrameworkScriptFile": "<rootDir>/tests/setup.js",
+    "setupFilesAfterEnv" : [
+      "<rootDir>/tests/setup.js"
+    ],
```
Comment (saturninoabril) [inline package.json]: 🎉

File: components/local_date_time/local_date_time.jsx (hunk 1 above)
Comment (lieut-data) [reply components/local_date_time/local_date_time.jsx]: I've also ripped out the snapshots for this particular component -- they really weren't adding any value here, and I've gone ahead and tested the parts that matter explicitly (namely the `title`).

File: package.json (hunk 3 above)
Comment (bradjcoughlin) [reply package.json]: Why the change here @lieut-data @saturninoabril ?

Comment (lieut-data) [reply package.json]: @bradjcoughlin, this resolves a depreciation warning emitted by Jest. Turns out `setupTestFrameworkScriptFile` only supports one file, so Jest is moving to `setupFilesAfterEnv`.

**Pull request merged**